packages/net/hass-tracker/files/functions.sh

```diff
@@ -50,23 +53,53 @@ build_payload() {
     host=$2
     consider_home=$3
     source_name=$4
-    
+
     echo "{\"mac\":\"$mac\",\"host_name\":\"$host\",\"consider_home\":\"$consider_home\",\"source_type\":\"router\",\"attributes\":{\"source_name\":\"$source_name\"}}"
 }
 
 get_ip() {
+    ret=$(get_ip_arp $@)
+    [ -z "$ret" ] && ret=$(get_ip_dhcp $@)
+    echo $ret
+}
+
+get_ip_arp() {
     # get ip for mac
-    grep "0x2\s\+$1" /proc/net/arp | cut -f 1 -s -d" "
+    grep "0x2\s\+$1" /proc/net/arp | cut -f 1 -s -d" " | grep -v '^169.254'
+}
+
+get_ip_dhcp() {
+    # get ip from dhcp table
+    leasefile="$(uci get dhcp.@dnsmasq[0].leasefile)"
+    grep "$1" "$leasefile" | cut -f 3 -s -d" "
 }
 
 get_host_name() {
+    ret=$(get_host_name_dhcp $@)
+    [ -z "$ret" ] && ret=$(get_host_name_dns $@)
+    echo $ret
+}
+
+get_host_name_dns() {
     # get hostname for mac
-    nslookup "$(get_ip $1)" | grep -o "name = .*$" | cut -d ' ' -f 3
+    domain="$(uci get dhcp.@dnsmasq[0].domain)"
+    nslookup "$(get_ip $1)" | grep -o "name = .*$" | cut -d ' ' -f 3 | sed -e "s/\\.$domain//"
```

Comment:
Why would you strip the domain?

Reply:
It would be better to specify the MAC Address instead of the hostname. If you have multiple AP running OpenWrt plus 802.11r fast roaming, the hostname is irrelevant for the AP, only the MAC is used.

Fri Jan 31 11:37:09 2020 daemon.notice hostapd: wlan1: AP-STA-DISCONNECTED fc:18:3c:xx:xx:xx
Fri Jan 31 11:37:09 2020 daemon.info hostapd: wlan1: STA fc:18:3c:xx:xx:xx IEEE 802.11: authenticated
Fri Jan 31 11:37:09 2020 daemon.info hostapd: wlan1: STA fc:18:3c:xx:xx:xx IEEE 802.11: associated (aid 4)